src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { st, classes } from './Dialog.st.css';
+import { DATA_HOOKS } from './constants';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+
+import { Modal } from '../Modal';
+import { IconButton } from '../IconButton';
+import { Close as CloseIcon } from '../../assets/icons';
+import { TPAComponentProps } from '../../types';
+import { KEY_CODES } from '../../common/keyCodes';
+
+export interface DialogProps extends TPAComponentProps {
+  /** Whether the modal is opened */
+  isOpen?: boolean;
+  /** Whether to use our focus trap or to manage the focus manually. */
+  manualFocus?: boolean;
+  /** Adds a ref for the close button. This prop might be used when you choose the manual focus. */
+  closeButtonRef?: React.RefObject<HTMLButtonElement>;
+  /** Callback for when the close button is clicked */
+  onClose?(): void;
+  /** Children to render inside dialog */
+  children?: React.ReactNode;
+  /** Defines a string value that labels the dialog element. Optional. */
+  'aria-label'?: string;
+  /** Identifies the element that labels the dialog element. Optional. */
+  'aria-labelledby'?: string;
+  /** Gives the dialog an accessible description by referring to the dialog content that describes the primary message or purpose of the dialog. Optional. */
+  'aria-describedby'?: string;
+  /** Defines a string value that labels the close button element. Optional. */
+  'close-button-aria-label'?: string;
+  /** Identifies the element that labels the close button element. Optional. */
+  'close-button-aria-labelledby'?: string;
```

Comment:
I'm not sure about the naming here.
Usually we use `aria-label` and such, in order to align with native html.
Since `aria-label` and `data-` attributes exist in html and React, we prefer to use them as is.
the close-button-aria-label, however, isn't a native html attribute.
So in this case I think we can wither have the prop named: `closeButtonAriaLabel`
or have a prop: `closeButtonAria` prop that accepts an object:
```ts
{
  label: string,
  labelledby: string,
}
```

wdyt?

Reply:
I also wasn't sure what will be a good naming for it.I change for "closeButtonAriaLabel"